### ISSUE TITLE
undefined symbol: simd_alignment with gcc 5.3.0

### ISSUE THREAD

**Issue by kshpytsya** [pyFFTW/pyFFTW#89]
```
# CC=gcc-4.9.3 pip install --no-cache-dir --upgrade --force pyfftw
# nm -D $HOME/.local/lib/python2.7/site-packages/pyfftw/pyfftw.so|grep simd_alignment
# python -c 'import pyfftw'
# CC=gcc-5.3.0 pip install --no-cache-dir --upgrade --force pyfftw
# nm -D $HOME/.local/lib/python2.7/site-packages/pyfftw/pyfftw.so|grep simd_alignment
         U simd_alignment
# python -c 'import pyfftw'
ImportError: /home/.../.local/lib/python2.7/site-packages/pyfftw/pyfftw.so: undefined symbol: simd_alignment

32-bit gentoo linux
python 2.7.10
cython 0.23.4
numpy 1.10.1
```

**Comment by hgomersall** [comment]
Interesting.

What compile flags were used? Can you show me the output of the compile process?

**Comment by hgomersall** [comment]
Just further to this, I really need all the compile command (and its output) which should be outputted by downloading the package and running `python setup.py build_ext`. If I can see all the compile steps then hopefully I'll be able to reproduce the problem.

**Comment by kshpytsya** [comment]
It essentially boils down to:

```
# gcc-5.3.0 -fPIC -I/home/uken/src-foreign/pyFFTW-0.10.1/include -I/home/uken/src-foreign/pyFFTW-0.10.1/pyfftw -I/usr/lib/python2.7/site-packages/numpy/core/include -I/usr/include/python2.7 -c /home/uken/src-foreign/pyFFTW-0.10.1/pyfftw/pyfftw.c -o build/temp.linux-i686-2.7/home/uken/src-foreign/pyFFTW-0.10.1/pyfftw/pyfftw.o
# i686-pc-linux-gnu-gcc -pthread -shared -Wl,-O1 -Wl,--as-needed -L. build/temp.linux-i686-2.7/home/uken/src-foreign/pyFFTW-0.10.1/pyfftw/pyfftw.o -L/usr/lib -lfftw3 -lfftw3f -lfftw3l -lfftw3_threads -lfftw3f_threads -lfftw3l_threads -lpython2.7 -o build/lib.linux-i686-2.7/pyfftw/pyfftw.so
```
With gcc-5.3.0 there is an undefined "simd_alignment" in pyfftw.so, with gcc-4.9.3 there isn't.

**Comment by hgomersall** [comment]
I think this is fixed with PR #90. It would be good if you could do a quick code review and test the resultant output.

FYI I could reproduce the problem from your gcc commands, and remove it with any level of optimisation. The problem was the wrong semantics were being used for declaring the inline `simd_alignment` function in the relevant header. Making it `static inline` seems to do the trick (which is what is intended).

Thanks for the bug report and your helpful and quick responses.

**Comment by hgomersall** [comment]
Another FYI: The problem is not specific to GCC 5.3.

**Comment by kshpytsya** [comment]
Well, I am not sure if I am in position to do code reviews for this project. However I can confirm that PR #90 fixes this issue.

**Comment by hgomersall** [comment]
No probs. I've merged PR #90 into master. I'll make another release when debian improvements are merged.
